src/stores/AppStore.js

```diff
@@ -727,15 +726,20 @@ export default types
     }
 
     function setHistory(history = []) {
+      console.log('Setting annotation history');
       const as = self.annotationStore;
 
+      const selected = as.selected?.toJSON();
+
+      console.log({ pk: selected?.pk, id: selected?.id });
+
       as.clearHistory();
 
       // always check that history is for correct and submitted annotation
-      if (!history.length || !as.selected?.pk) return;
-      if (Number(as.selected.pk) !== Number(history[0].annotation_id)) return;
+      if (Array.isArray(history) && !history.length || !as.selected?.pk) return console.warn('Invalid history record', history);
```

Comment:
why do we need check for array here? could it be something else and why?
we use it as array 2 lines down with no checks

Reply:
it's js, it could be anything :)

Reply:
I assume at some point we were getting "cannot read length of undefined" or something like that. Honestly, I don't remember already why I did that. Shouldn't be a problem though. At least this code shows that we expect an array and nothing else.